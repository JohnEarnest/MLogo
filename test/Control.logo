# Control
#
# test primitive control structures
# and their side-effects on scopes.

# simple execution
print 'Before
run [
	print 'Hello,
	print 'World!
]
print 'After

# simple conditionals
if     'true  [ print 'tyep  ]
unless 'true  [ print 'tnope ]
if     'false [ print 'fyep  ]
unless 'false [ print 'fnope ]

# simple repeat
repeat 4 [ print 999 print 666 ]

# nested repeat
repeat 3 [
	print 'a
	repeat 2 [ print 'b ]
]

# make sure 'run' pushes a scope:
local 'x 23
run [
	local 'x 64
	print :x
]
print :x

# make sure 'if and 'unless push scopes:
if     'true  [ local 'x 93 print :x ]
unless 'false [ local 'x 39 print :x ]
print :x