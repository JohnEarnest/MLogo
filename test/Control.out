'Before
'Hello,
'World!
'After
'tyep
'fnope
999
666
999
666
999
666
999
666
'a
'b
'b
'a
'b
'b
'a
'b
'b
64
23
93
39
23
